clang/include/clang/Serialization/ASTReader.h

```diff
@@ -1392,6 +1392,20 @@ class ASTReader
 
   llvm::DenseMap<const Decl *, bool> DefinitionSource;
 
+  /// The set of extra flags about declarations that we have read from
+  /// the module file.
+  struct ExternalDeclarationBits {
+    /// Indicates if given function declaration was a definition but its body
+    /// was removed due to declaration merging.
+    bool ThisDeclarationWasADefinition : 1;
+
+    ExternalDeclarationBits() : ThisDeclarationWasADefinition(false) {}
+  };
+
+  /// A mapping from declarations to extra bits of information about this decl.
+  llvm::DenseMap<const Decl *, ExternalDeclarationBits>
```

Comment:
```suggestion
  llvm::DenseMap<const Decl *, bool>
```

nit: I think we can avoid the abstraction here.

Reply:
In this case we don't even need map, I replaced with with a set.